### ISSUE TITLE
[debops.ldap] missing some local fact variables w/ Python 3

### ISSUE THREAD

**Issue by imrejonk** [debops/debops#1071]
Using DebOps v1.1.1, Ansible 2.7 and ~~Debian 10~~ **Debian 9**

Running the `ldap.fact` script with the Python 2.7 interpreter works flawlessly:
```
{
    "base_dn": [
        "dc=ciphermail", 
        "dc=net"
    ], 
    "basedn": "dc=ciphermail,dc=net", 
    "conf": [
        {
            "base": "dc=ciphermail,dc=net"
        }, 
        {
            "uri": "ldap://0.ldap.ciphermail.net/ ldap://1.ldap.ciphermail.net/"
        }, 
        {
            "tls_cacert": "/etc/ssl/certs/ca-certificates.crt"
        }, 
        {
            "tls_reqcert": "demand"
        }
    ], 
    "configured": true, 
    "device_dn": [
        "cn=wildfire.ciphermail.net", 
        "dc=ciphermail.net", 
        "ou=Hosts", 
        "dc=ciphermail", 
        "dc=net"
    ], 
    "enabled": true, 
    "groupid_max": "2001999999", 
    "groupid_min": "2000000000", 
    "groups_rdn": "ou=Groups", 
    "home": "/home", 
    "hosts": [
        "0.ldap.ciphermail.net", 
        "1.ldap.ciphermail.net"
    ], 
    "hosts_rdn": "ou=Hosts", 
    "machines_rdn": "ou=Machines", 
    "people_rdn": "ou=People", 
    "port": "389", 
    "protocol": "ldap", 
    "services_rdn": "ou=Services", 
    "shell": "/bin/bash", 
    "start_tls": true, 
    "system_groups_rdn": "ou=System Groups", 
    "tls_cacert": "/etc/ssl/certs/ca-certificates.crt", 
    "tls_reqcert": "demand", 
    "uid_gid_max": "2099999999", 
    "uid_gid_min": "2000000000", 
    "uri": [
        "ldap://0.ldap.ciphermail.net/", 
        "ldap://1.ldap.ciphermail.net/"
    ]
}
```
(yep, we have an OpenLDAP cluster thanks to @drybjed's awesome work on the `debops.slapd` role and documentation)

However, running the same script with ~~Python 3.7~~ **Python 3.5** yields less results:
```
{
    "configured": false,
    "device_dn": [
        "cn=wildfire.ciphermail.net",
        "dc=ciphermail.net",
        "ou=Hosts",
        "dc=ciphermail",
        "dc=net"
    ],
    "enabled": true,
    "groupid_max": "2001999999",
    "groupid_min": "2000000000",
    "groups_rdn": "ou=Groups",
    "home": "/home",
    "hosts_rdn": "ou=Hosts",
    "machines_rdn": "ou=Machines",
    "people_rdn": "ou=People",
    "services_rdn": "ou=Services",
    "shell": "/bin/bash",
    "system_groups_rdn": "ou=System Groups",
    "uid_gid_max": "2099999999",
    "uid_gid_min": "2000000000"
}
```
It looks like the script fails to append data from /etc/ldap/ldap.conf.

**Comment by drybjed** [comment]
@imrejonk I tried to reproduce the issue in my environment, but the script works fine with Python 3.7 installed. Are you sure that the problem is still present? Can you renmove the `try/except` clauses in the script to see what errors show up when you run the script?

**Comment by imrejonk** [comment]
Ah, my bad. The machine I tested this on was running Debian 9 with Python 3.5 instead of Debian 10 with Python 3.7, which is what we have running on all our other servers. That machine is still on Debian 9 because of MongoDB and OpenJDK 8 needed for our Unifi Controller.

I removed the `try/except` clauses, turns out Python 3.5 in Debian 9 doesn't have an `ldap` module available:
```
imre@unifi:/etc/ansible/facts.d$ python3 ldap-test.fact
Traceback (most recent call last):
  File "./ldap-test.fact", line 9, in <module>
    import ldap
ImportError: No module named 'ldap'
```

It's also not available in the Debian 9 repositories:
```
imre@unifi:/etc/ansible/facts.d$ apt policy python3-ldap
python3-ldap:
  Installed: (none)
  Candidate: (none)
  Version table:
```

Python 2 support is disabled by default during a run of the bootstrap playbook. I noticed the missing variable problem when running the `debops.sudo` role, which fails on the LDAP setup because of the missing variables.

Not sure what's the best course of action here. We could:

1. Close this issue and only support Debian stable
2. Not disable Python 2 on Debian 9
3. Find some workaround for this issue on Debian 9

**Comment by drybjed** [comment]
Hmmm, Python 2 should be eanbled by default on Debian <10, let me check the code...

**Comment by drybjed** [comment]
@imrejonk This might have been a bug in raw Python version detection code, fixed by https://github.com/debops/debops/pull/1073. Can you check if with this fix you still have Python 2.7 disabled on Debian 9? It should be enabled there.

**Comment by imrejonk** [comment]
@drybjed that did it! Just ran the `bootstrap` playbook from master against a fresh Debian 9 VM. These Python packages are installed:

```
imre@devimre:~$ dpkg -l | awk '{print $2}' | grep '^python'
python
python-all
python-all-dev
python-apt
python-apt-common
python-cffi-backend
python-crypto
python-cryptography
python-dbus
python-dev
python-dnspython
python-enum34
python-gi
python-httplib2
python-idna
python-ipaddress
python-keyring
python-keyrings.alt
python-minimal
python-pip
python-pip-whl
python-pkg-resources
python-pyasn1
python-pycurl
python-secretstorage
python-setuptools
python-six
python-virtualenv
python-wheel
python-xdg
python2.7
python2.7-dev
python2.7-minimal
python3
python3-apt
python3-cffi-backend
python3-crypto
python3-cryptography
python3-dbus
python3-dev
python3-dnspython
python3-gi
python3-httplib2
python3-idna
python3-keyring
python3-keyrings.alt
python3-minimal
python3-pip
python3-pkg-resources
python3-pyasn1
python3-pycurl
python3-secretstorage
python3-setuptools
python3-six
python3-virtualenv
python3-wheel
python3-xdg
python3.5
python3.5-dev
python3.5-minimal
```

The `debops.ldap`, `debops.nslcd` and `debops.sudo` roles also work beautifully :)
